### PR TITLE
adds deep coverage zone routed percentage to TP's routing widget

### LINKED ISSUE #3752: deep czf routing method missing from TP dashboard
the TP dashboard has a widget to display routing methods, however, the deep czf routing method is not represented:

image

routing method percentages are derived from the following api:

`GET /api/1.4/cdns/routing`

### PR DESCRIPTION
**Pull request by mitchell852**
## What does this PR (Pull Request) do?

- [x] This PR fixes #3752 

Note: the TP dashboard is not currently under test.

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?
1. Run the UI tests and ensure they pass
2. Navigate to the TP dashboard and view the routing methods widget and ensure there is an entry for "Deep Coverage Zone" with a percentage that lines up with GET /api/1.4/cdns/routing

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_portal/app/src/common/modules/widget/routing/widget.routing.tpl.html [hunk 1]
```diff
@@ -36,6 +36,20 @@ <h2>Routing Methods</h2>
         </div>
         <div class="clearfix"></div>
     </div>
+    <div class="widget_summary">
+        <div class="w_left w_25">
+            <span>Deep Coverage Zone</span>
+        </div>
+        <div class="w_center w_55">
+            <div class="progress">
+                <div class="progress-bar" role="progressbar" ng-style="{'width': deepCoverageZone + '%'}" style="background-color: #3063BD;"></div>
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/widget/routing/widget.routing.tpl.html]: Why not use [the `<meter>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meter)?
It'd look like:
```html
<div class="widget_summary">
	<div class="w_left w_25">
		<label for="deepCZ-meter">Deep Coverage Zone</label>
	</div>
	<div class="w_center w_55">
		<meter min=0 max=100 ng-value="deepCoverageZone" id="deepCZ-meter">{{ ::deepCoverageZone | number:3}}%</meter>
	</div>
	<div class="clearfix"></div>
</div>
```

Which has the added benefit of a more appropriate `role` than `progressbar`, and adding proper labeling relationships.

Comment (mitchell852) [reply traffic_portal/app/src/common/modules/widget/routing/widget.routing.tpl.html]: > Why not use [the `<meter>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meter)?
> It'd look like:
> 
> ```
> <div class="widget_summary">
> 	<div class="w_left w_25">
> 		<label for="deepCZ-meter">Deep Coverage Zone</label>
> 	</div>
> 	<div class="w_center w_55">
> 		<meter min=0 max=100 ng-value="deepCoverageZone" id="deepCZ-meter">{{ ::deepCoverageZone | number:3}}%</meter>
> 	</div>
> 	<div class="clearfix"></div>
> </div>
> ```
> 
> Which has the added benefit of a more appropriate `role` than `progressbar`, and adding proper labeling relationships.

I gave that a shot. didn't really work so just going to stick with the current approach.

File: docs/source/api/cdns_routing.rst [hunk 2]
```diff
@@ -33,12 +33,16 @@ No parameters available
...
 Response Structure
 ------------------
-:cz:          Used Coverage Zone geographic IP mapping
-:dsr:         Overflow traffic sent to secondary CDN
-:err:         Error localizing client IP
-:geo:         Used 3rd party geographic IP mapping
-:miss:        No location available for client IP
-:staticRoute: Used pre-configured DNS entries
+:cz:                The percent of requests to the Traffic Router for this :term:`Delivery Service` that were satisfied by a :term:`Coverage Zone File`
+:deepCz:            The percent of requests to the Traffic Router for this :term:`Delivery Service` that were satisfied by a Deep :term:`Coverage Zone File`
```
Comment (ocket8888) [inline docs/source/api/cdns_routing.rst]: "Deep Coverage Zone File" is a `term` of its own.

File: docs/source/api/cdns_routing.rst [hunk 3]
```diff
@@ -33,12 +33,16 @@ No parameters available
...
+:dsr:               The percent of requests to the Traffic Router for this :term:`Delivery Service` that were satisfied by sending the client to an overflow :term:`Delivery Service`
+:err:               The percent of requests to the Traffic Router for this :term:`Delivery Service` that resulted in an error
+:fed:               The percent of requests to the Traffic Router for this :term:`Delivery Service` that were satisfied by sending the client to a federated CDN
+:geo:               The percent of requests to the Traffic Router for this :term:`Delivery Service` that were satisfied using 3rd party geographic IP mapping
```
Comment (ocket8888) [inline docs/source/api/cdns_routing.rst]: you could put the "rd" into superscript if you'd like. It's just a contraction, so there's nothing grammatically wrong with what you have here.

```rst
... 3\ :sup:`rd`...
```
renders like: 3<sup>rd</sup>

File: docs/source/api/deliveryservices_id_routing.rst [hunk 4]
```diff
@@ -49,6 +49,7 @@ Request Structure
 Response Structure
 ------------------
 :cz:                The percent of requests to the Traffic Router for this :term:`Delivery Service` that were satisfied by a :term:`Coverage Zone File`
+:deepCz:            The percent of requests to the Traffic Router for this :term:`Delivery Service` that were satisfied by a Deep :term:`Coverage Zone File`
```
Comment (ocket8888) [inline docs/source/api/deliveryservices_id_routing.rst]: same as above RE: "Deep Coverage Zone File"

File: traffic_portal/app/src/common/modules/widget/routing/widget.routing.tpl.html (hunk 1 above)
Comment (mitchell852) [reply traffic_portal/app/src/common/modules/widget/routing/widget.routing.tpl.html]: ok, it worked this time with a height and width on it

File: docs/source/api/cdns_routing.rst (hunk 3 above)
Comment (mitchell852) [reply docs/source/api/cdns_routing.rst]: i suppose i'll just leave it

**Pull request merged**